Modify EditBox for win32

summerinsects opened the pull request:
1. Do render when the system DialogBox popup
2. Star('*') wll be shown in win32 edit control if the EditBox's input flag is
   password
3. Function editBoxTextChanged will be called when the text in win32
   edit control is changed
4. The temporary variable (here is std::string) may destruct
   when use expression like AnsiToUtf8(...).c_str() or Utf8ToAnsi(...).c_str()

Commit:
Modify EditBox for win32

1. Do render when the system DialogBox popup
2. '*' wll be shown in win32 edit control if the EditBox's input flag is
password
3. Function editBoxTextChanged will be called when the text in win32
edit control is changed
4. Return value std::string is a temporary variable, that may destruct
when AnsiToUtf8(...).c_str() or Utf8ToAnsi(...).c_str()

zilongshanren commented on extensions/GUI/CCEditBox/CCEditBoxImplWin.cpp:
```diff
@@ -149,30 +151,39 @@ void EditBoxImplWin::setText(const char* pText)
 {
     if (pText != nullptr)
     {
+        if (_text.compare(pText))
```
Do you mean _text equals to pText, then we should return?
It think here should be (0 == _text.compare(pText)).

summerinsects replied:
I'm so sorry
Yes. It should return if _text equals to pText
Btw, bool EditBoxImplWin::isEditing() should return _isEditing
Thx!

summerinsects replied:
I'm sorry than in function char\* CWin32InputBox::AnsiToUtf8(const char\* strAnsi) the local variable ret has not been initialized, and NULL should be assigned

Commit:
fixed some error

Commit:
Merge pull request #2 from cocos2d/v3

update

Commit:
set FILE_SHARE_READ to the 3rd parameter(dwShareMode) when call CreateFileW

Otherwise, in cpp-tests some of the fonts may not be displayed
correctly.
The log is as follows:
Get data from file(fonts/arial.ttf) failed, error code is 32

Commit:
Revert "set FILE_SHARE_READ to the 3rd parameter(dwShareMode) when call CreateFileW"

This reverts commit a327936725743e13ecc7b026c6ed43758619e94f.

Commit:
fixed: the label cannot be displayed in cpp-tests for platform win32 sometimes

fixed: the label cannot be displayed in cpp-tests for platform win32
sometimes
Pass FILE_SHARE_READ to the 3rd parameter of CreateFileW. Otherwise, it
may fail to get date from the font file.

Commit:
Revert "fixed some error"

This reverts commit fd1184b80a32301c72b83e57bf74b5b69b4a0c7a.

Commit:
Revert "Modify EditBox for win32"

This reverts commit 1c3868e8bf87ce8fd636ee38404d12e4cd286ae2.
